### PR TITLE
Add PyTorch Ignite example.

### PR DESCRIPTION
**Pull request by toshihikoyanase**
This PR adds an example of [PyTorch Ignite](https://pytorch.org/ignite/). The training code was created based on [the MNIST example](https://github.com/pytorch/ignite/blob/master/examples/mnist/mnist.py) in the Ignite repository.

### CODE REVIEW

File: examples/ignite_simple.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    model = Net(trial)
+    device = 'cpu'
+    if torch.cuda.is_available():
+        device = 'cuda'
+
+    # We optimize the learning rate and momentum factor of optimizer.
+    lr = trial.suggest_loguniform('lr', 1e-7, 1e-1)
+    momentum = trial.suggest_uniform('momentum', 0.1, 0.9)
+
+    optimizer = SGD(model.parameters(), lr=lr, momentum=momentum)
```
Comment (sile) [inline examples/ignite_simple.py]: How about suggesting optimizer's type?
(e.g., [chainer_simple.py#L55](https://github.com/pfnet/optuna/blob/master/examples/chainer_simple.py#L55))

Comment (toshihikoyanase) [reply examples/ignite_simple.py]: Sounds good! I'll fix it.

Comment (toshihikoyanase) [reply examples/ignite_simple.py]: Fixed: https://github.com/pfnet/optuna/pull/559/commits/313113b67908df5095d6502b9dfba765214253ae

File: examples/ignite_simple.py [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+        x = F.relu(F.max_pool2d(self.conv1(x), 2))
+        x = F.relu(F.max_pool2d(self.conv2_drop(self.conv2(x)), 2))
+        x = x.view(-1, 320)
+        x = F.relu(self.fc1(x))
+        x = F.dropout(x, training=self.training)
+        x = self.fc2(x)
+        return F.log_softmax(x, dim=-1)
+
+
+def create_optimizer(trial, model):
```
Comment (smly) [inline examples/ignite_simple.py]: (Optional) In my experience, tuning the learning rate makes little sense in a simple binary classification task with a single machine. `optimizer = Adam(model.parameters(), lr=1e-3)` is good enough.

File: examples/ignite_simple.py [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+class Net(nn.Module):
+    def __init__(self, trial):
+        # We optimize dropout rate in a convolutional neural network.
+        super(Net, self).__init__()
+        self.conv1 = nn.Conv2d(1, 10, kernel_size=5)
+        self.conv2 = nn.Conv2d(10, 20, kernel_size=5)
+        dropout_rate = trial.suggest_uniform('dropout_rate', 0, 1)
+        self.conv2_drop = nn.Dropout2d(p=dropout_rate)
+        self.fc1 = nn.Linear(320, 50)
```
Comment (smly) [inline examples/ignite_simple.py]: (Optional) I suggest tuning the size of features in a FC layer.

```py
fc2_input_dim = trial.suggest_int('fc2_input_dim', 40, 80)
self.fc1 = nn.Linear(320, fc2_input_dim)
self.fc2 = nn.Linear(fc2_input_dim, 10)
```

File: examples/ignite_simple.py (hunk 2 above)
Comment (toshihikoyanase) [reply examples/ignite_simple.py]: That makes sense.
According to [the reference](https://pytorch.org/docs/stable/_modules/torch/optim/adam.html#Adam), `lr=1e-3` seems to be the default value.
I'll change it to `optimizer = Adam(model.parameters())`.

On the other hand, I'll keep the learning rate tuning for [`SGD`](https://pytorch.org/docs/stable/optim.html#torch.optim.SGD) because the learning rate is a required parameter.

File: examples/ignite_simple.py (hunk 3 above)
Comment (toshihikoyanase) [reply examples/ignite_simple.py]: Sounds good!
I'll fix it. Thank you.

File: examples/ignite_simple.py (hunk 2 above)
Comment (toshihikoyanase) [reply examples/ignite_simple.py]: I'll remove `SGD` from the choices because it rarely outperformed the default setting of `Adam`.
